polyglot/piranha/src/models/source_code_unit.rs

```diff
@@ -72,12 +80,59 @@ impl SourceCodeUnit {
 
   pub(crate) fn apply_edit(&mut self, edit: &Edit, parser: &mut Parser) -> InputEdit {
     // Get the tree_sitter's input edit representation
-    self._apply_edit(
+    let mut applied_edit = self._apply_edit(
       edit.replacement_range(),
       edit.replacement_string(),
       parser,
       true,
-    )
+    );
+    // Check if the edit kind is "DELETE something"
+    if edit.replacement_string().is_empty() {
+      let deleted_at = edit.replacement_range().start_point.row;
+      if let Some(comment_range) = self.get_comment_at_line(deleted_at, 2) {
```

Comment:
Let's make the magic constant `2` some named constant somewhere, at least, if not a configurable parameter.